_appmap/test/test_generation.py

```diff
@@ -48,3 +53,18 @@ def check_comment(self, to_dict):
             return ret
 
         verify_example_appmap(check_comment, "instance_method")
+
+class TestAppMapEncoder:
```

Comment:
This test is good. I think it would also be helpful, especially to understand the impact of the change, to update one of the tests that actually generates an AppMap. One of these two could be good: https://github.com/getappmap/appmap-python/blob/e4b8d78638e97527d30605643ddd2903aa063d55/_appmap/test/web_framework.py#L286, or https://github.com/getappmap/appmap-python/blob/master/_appmap/test/test_test_frameworks.py#L94

Reply:
Actually, I tried to create exactly this as a separate AppMap generation test but wasn't able to produce the expected error - test passes without the fix as well. Returning int64 typed values from or accepting them as arguments to recorded functions seems not to produce the error. 

Here is a temp branch if you want to take a look what am I missing there: https://github.com/getappmap/appmap-python/tree/temp/test-recording-with-numpy-int64

Reply:
It's hard to tell why this isn't reproducing the expected behavior, because it's only checking the length of the event array.

The reason I picked those other two test cases was because they'd requiring updating the expected AppMap data.

Rather than entirely new test case, can you try updating one of those tests I suggested, please?

Reply:
It was not possible to produce a non JSON serializable AppMap by simply introducing a `numpy.int64` value as an event parameter or as a return value of an event, since all values are converted to strings with `event.py::display_string` while encoding their container `Event`.

While examining the `seaborn` project which had this [problem](https://github.com/getappmap/appmap-python/issues/312), it turned out that the issue occurs when `numpy.int64` is used as a key in a `dict` that is a parameter or a return value of a recorded function. These keys become the `name` attributes of properties while describing the value and the schema of the return value or the parameter. The `name` attributes are not checked or converted to strings in `event.py`, which leads to this error.

In addition to the direct tests for `AppMapEncoder`, I modified one of the existing tests you suggested to produce the AppMap that includes the recording of a function returning a `dict` with `numpy.int64` keys.